sockeye/data_io.py

```diff
@@ -411,6 +418,9 @@ def load(self,
                        for (source_len, target_len), num_samples in zip(self.buckets, num_samples_per_bucket)]
         data_target = [np.full((num_samples, target_len), self.pad_id, dtype=self.dtype)
                        for (source_len, target_len), num_samples in zip(self.buckets, num_samples_per_bucket)]
+        data_source_factor_sources = [[np.full((num_samples, source_len), self.pad_id, dtype=self.dtype)
```

Comment:
could we also move the factor data into `data_source`, making it 3d?
So without factors it would be `(num_samples, source_len, 1)`, and with n factors `(num_samples, source_len, 1+n)`.